LibVLCSharp.Forms/Shared/MediaPlayerElement.xaml.cs

```diff
@@ -0,0 +1,210 @@
+        {
+            var videoView = VideoView;
+            if (videoView != null)
+            {
+                videoView.MediaPlayer = mediaPlayer;
+            }
+            var playbackControls = PlaybackControls;
+            if (playbackControls != null)
+            {
+                playbackControls.MediaPlayer = mediaPlayer;
+            }
+        }
+
+        private void OnPlayControlsChanged(PlaybackControls playbackControls)
+        {
+            if (playbackControls != null)
+            {
+                playbackControls.MediaPlayer = MediaPlayer;
+                playbackControls.VideoView = VideoView;
+            }
+        }
+
+        private static void VideoViewPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            ((MediaPlayerElement)bindable).OnVideoViewChanged((VideoView)newValue);
+        }
+
+        private static void LibVLCPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            ((MediaPlayerElement)bindable).OnLibVLCChanged((LibVLC)newValue);
+        }
+
+        private static void MediaPlayerPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            ((MediaPlayerElement)bindable).OnMediaPlayerChanged((LibVLCSharp.Shared.MediaPlayer)newValue);
+        }
+
+        private static void PlaybackControlsPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            ((MediaPlayerElement)bindable).OnPlayControlsChanged((PlaybackControls)newValue);
+        }
+
+        /// <summary>
+        /// Invoked whenever the <see cref="Element.Parent"/> of an element is set. 
+        /// Implement this method in order to add behavior when the element is added to a parent.
+        /// </summary>
+        /// <remarks>Implementors must call the base method.</remarks>
+        protected override void OnParentSet()
+        {
+            base.OnParentSet();
+
+            if (Parent != null && !Initialized)
+            {
+                Initialized = true;
+
+                if (VideoView == null)
+                {
+                    VideoView = new VideoView();
+                }
+                if (PlaybackControls == null)
+                {
+                    PlaybackControls = new PlaybackControls();
+                }
+
+                var application = Application.Current;
+                application.PageAppearing += PageAppearing;
+                application.PageDisappearing += PageDisappearing;
+            }
+        }
+
+        private void PageAppearing(object sender, Page e)
+        {
+            if (e == this.FindAncestor<Page>())
+            {
+                MessagingCenter.Subscribe<LifecycleMessage>(this, "OnSleep", m =>
+                {
+                    var applicationProperties = Application.Current.Properties;
+                    applicationProperties["VLC_MediaPlayerElement_Position"] = MediaPlayer?.Position;
```

Comment:
I often had more problems by defining the time rather than the position. That's why I always use the position. But, I have not retested for a long time, it would be interesting to know what is better.

Reply:
Let's try it as it is for now and wait for feedback.